lib/strong_migrations/migration_helpers.rb

```diff
@@ -73,16 +75,172 @@ def remove_null_constraint_safely(table_name, column_name, name: nil)
       end
     end
 
+    def add_column_safely(table_name, column_name, type, **options)
+      ensure_postgresql(__method__)
+      ensure_not_in_transaction(__method__)
+
+      default = options.delete(:default)
+
+      if postgresql_version >= 110000 || default.nil?
+        add_column(table_name, column_name, options)
+      else
+        reversible do |dir|
+          dir.up do
+            transaction do
+              add_column(table_name, column_name, type, default: nil, **options)
```

Comment:
Should this only add the column if it doesn't already exist to properly handle retries since the entire transaction isn't wrapped in a transaction? We've hit issues with longer running migrations so we generally try to make them idempotent.

Reply:
@jturkel Thanks for your suggestions. Implemented them in #105